src/main/java/com/rackspace/jenkins_nodepool/NodePoolJobHistory.java

```diff
@@ -0,0 +1,65 @@
+package com.rackspace.jenkins_nodepool;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Track historical information about Jenkins jobs handled by the plugin.
+ */
+public class NodePoolJobHistory implements Iterable<NodePoolJob> {
+
+    private static final int MAX_HISTORY_LENTH = 100;
+    private final Object lock = new Object();
+
+    private final List<NodePoolJob> jobs;
+    private final int maxHistoryLenth;
+
+    public NodePoolJobHistory() {
+        this(MAX_HISTORY_LENTH);
+    }
+
+    public NodePoolJobHistory(final int maxHistoryLenth) {
+        this.maxHistoryLenth = maxHistoryLenth;
+        jobs = new ArrayList<>(maxHistoryLenth);
```

Comment:
Any need to check the bounds on this guy (`maxHistoryLength`)? What if it's larger than expected (e.g. greater than say 100 or 1000)?

Reply:
It's not exposed in the UI anywhere so we don't really need to bounds check it.  I plan to use it in a unit test though.